test/v2/unit/ethWallet.ts

```
@@ -46,6 +51,234 @@ describe('Sign ETH Transaction', co(function *() {
+  };
+
+  describe('Verify HSM Hop prebuild', co(function *() {
+    let prebuild;
+    let buildParams;
+    let finalRecipient;
+    let sendAmount;
+
+    before(co(function *() {
+      finalRecipient = '0x5208d8e80c6d1aef9be37b4bd19a9cf75ed93dc8';
+      sendAmount = '200000000000000';
+      prebuild = {
+        tx,
+        id: txid,
+        signature: bitgoSignature,
+      };
+      buildParams = {
+        recipients: [{
+          address: finalRecipient,
+          amount: sendAmount
+        }]
+      };
+    }));
+
+    it('should accept a valid hop prebuild', co(function *() {
+      let error = undefined;
+      nockBitGoKey();
+      try {
+        yield ethWallet.baseCoin.validateHopPrebuild(ethWallet, prebuild, buildParams);
+      } catch (e) {
+        error = e.message;
+      }
+      should.not.exist(error);
+    }));
+
+    it('should fail if the recipient is wrong', co(function *() {
+      let error = undefined;
+      const badBuildParams = JSON.parse(JSON.stringify(buildParams));
+      badBuildParams.recipients[0].address = '0x54bf1609aeed804aa231f08c53dbb18f7d374615';
+
+      nockBitGoKey();
+      try {
+        yield ethWallet.baseCoin.validateHopPrebuild(ethWallet, prebuild, badBuildParams);
+      } catch (e) {
+        error = e.message;
+      }
+      should.exist(error);
+      error.should.containEql("does not equal original recipient");
+    }));
+
+    it('should fail if the amount is wrong', co(function *() {
+      let error = undefined;
+      const badBuildParams = JSON.parse(JSON.stringify(buildParams));
+      badBuildParams.recipients[0].amount = '50000000';
+
+      nockBitGoKey();
+      try {
+        yield ethWallet.baseCoin.validateHopPrebuild(ethWallet, prebuild, badBuildParams);
+      } catch (e) {
+        error = e.message;
+      }
+      should.exist(error);
+      error.should.containEql("does not equal original amount");
+    }));
+
+    it('should fail if the signature is invalid', co(function *() {
```

Unclear. "should fail if the HSM signature is invalid"?